[ref:add-fluid-buffer-to-pylon-fluid-block] Fluid refactor

LordIdra opened the pull request:
Companion PR to https://github.com/pylonmc/pylon-core/pull/181

LordIdra commented on src/main/java/io/github/pylonmc/pylon/base/content/machines/fluid/PortableFluidTank.java:
```diff
@@ -152,67 +141,26 @@ public void write(@NotNull PersistentDataContainer pdc) {
+    @Override
+    public boolean setFluid(double amount) {
+        boolean result = PylonFluidTank.super.setFluid(amount);
+        float scale = (float) (0.9F * fluidAmount() / capacity);
```
That's intended, you wouldn't want to see any fluid in your portable fluid tank if there isn't any fluid

OhmV-IR replied:
ah got confused and thought it was the scale of the whole item

LordIdra commented:
> LGTM, going to let this go through without the events I mentioned previously since it isn't really the job of this PR to fix, but I'll mark this with a TODO gh issue

I'll address those, it doesn't make sense to make an issue for this, this is exactly the kind of thing we have reviews for. Just got distracted before addressing those comments

LordIdra commented:
it's kinda unclear what events need to be called but these two are probably sufficient and I can't imagine it'll matter much anyway

Seggan reviewed (approved):
Have I looked at any of this code? No. Am I gonna approve it anyway? Yes.
